excretor/src/routing.rs

```diff
@@ -159,4 +214,81 @@ mod handlers {
             ))
         }
     }
+
+    pub(super) async fn auth(
+        State(state): State<RouterState>,
+    ) -> Result<(StatusCode, Response), AppError> {
+        let scopes = "channels:read";
+        let slack_auth_url = format!(
+            "https://slack.com/oauth/v2/authorize?client_id={}&scope={}&redirect_uri={}",
+            state.env_vars.slack_client_id, scopes, state.env_vars.slack_redirect_uri
+        );
+
+        Ok((
+            StatusCode::FOUND,
+            Response::builder()
+                .header("Location", slack_auth_url)
+                .body(Body::empty())
+                .unwrap(),
+        ))
+    }
+
+    // get request with params
+    pub(super) async fn auth_callback(
+        State(state): State<RouterState>,
+        Query(request): Query<AuthCallback>,
+    ) -> Result<(StatusCode, Response), AppError> {
+        let code = request.code;
+        let slack_auth_url = format!(
+            "https://slack.com/api/oauth.v2.access?client_id={}&client_secret={}&code={}&redirect_uri={}",
+            state.env_vars.slack_client_id, state.env_vars.slack_client_secret, code, state.env_vars.slack_redirect_uri
+        );
+        // request slack for access token
+        let response = Client::new().get(slack_auth_url).send().await?;
+        
+        if response.status() != StatusCode::OK {
+            return Ok((
+                StatusCode::UNAUTHORIZED, 
+                Body::from("Mortals are forbidden from accessing the site").into_response()
```

Comment:
Store this string somewhere since it is being reused

Reply:
Done